vpr/src/route/rr_graph_indexed_data.cpp

```diff
@@ -264,101 +250,67 @@ static std::vector<size_t> count_rr_segment_types() {
     return rr_segment_type_counts;
 }
 
-static float get_delay_normalization_fac(int nodes_per_chan,
-                                         const t_rr_node_indices& L_rr_node_indices) {
+static float get_delay_normalization_fac() {
     /* Returns the average delay to go 1 CLB distance along a wire.  */
 
-    const int clb_dist = 3; /* Number of CLBs I think the average conn. goes. */
-
-    int inode, itrack, cost_index;
-    float Tdel, Tdel_sum, frac_num_seg;
-
     auto& device_ctx = g_vpr_ctx.device();
+    auto& rr_indexed_data = device_ctx.rr_indexed_data;
 
-    Tdel_sum = 0.;
+    std::vector<float> Tdel_vector(0.0);
+    for (size_t cost_index = CHANX_COST_INDEX_START; cost_index < rr_indexed_data.size(); cost_index++) {
+        float T_value = rr_indexed_data[cost_index].T_linear + rr_indexed_data[cost_index].T_quadratic;
 
-    for (itrack = 0; itrack < nodes_per_chan; itrack++) {
-        inode = find_average_rr_node_index(device_ctx.grid.width(), device_ctx.grid.height(), CHANX, itrack,
-                                           L_rr_node_indices);
-        if (inode == -1)
-            continue;
-        cost_index = device_ctx.rr_nodes[inode].cost_index();
-        frac_num_seg = clb_dist * device_ctx.rr_indexed_data[cost_index].inv_length;
-        Tdel = frac_num_seg * device_ctx.rr_indexed_data[cost_index].T_linear
-               + frac_num_seg * frac_num_seg
-                     * device_ctx.rr_indexed_data[cost_index].T_quadratic;
-        Tdel_sum += Tdel / (float)clb_dist;
+        if (rr_indexed_data[cost_index].number_of_nodes == 0 || T_value == 0.0) continue;
+
+        Tdel_vector.push_back(T_value * 1e10);
```

Comment:
Where did this magic number of come from?  It should be `constexpr double kNameOfConstant = 1e10`, along with a comment about the value should be.

Reply:
Right, I forgot to fix that. Anyway, this was needed to temporary shift delay values, as during the geo mean computation, the subsequent multiplication of the delays would eventually bring the `mult` value to zero. 
Unsure whether this is the best way to achieve this though.

Reply:
Floating point doesn't need shifts like this.  Rather than using 1e10, just a `double` during the intermediate calcuation.

Reply:
Actually I tried with a double for the intermediate calculation, but it went actually beyond the limit which is around `1e-300`

Reply:
So the issue is that maybe `product(Tdel_vector)` is the wrong math to use here?  I think you may have a numerical convergence error here.

Reply:
The fact is that I wanted to use the geometric mean to take into account possible delay values outliers of some segment types which, I have seen, may differ by two orders of magnitude.

Given that delay values are usually in the order of 1e-10, I applied the 1e10 correction for the intermediate step and than restored it at the end of the geo mean calculation.

Reply:
The solution here is to change from `(product(T_del))^(1/len(T_del))` to https://en.wikipedia.org/wiki/Geometric_mean#Relationship_with_logarithms `e^(1/len(T_del)*sum(map(logn(T_del))))`.   `logn(0)` is a violation, but those need to be skipped in either case.  Again, the problem here is the naive geomean algorithm converges always to zero as N increases.  This is numerically denegenerate, and you can always increase N to drive the product to zero.

Reply:
I don't think geometric mean is the right approach here.  It does fairly average numers with widely varying magnitude ranges (treats them all as equally significant) but it is very dangerous if numbers can go to or approach 0.  We could get some small/weird wires or fragments with delay near 0, and we really do not want those dropping the average way down.  I think that's why you're special casing 0 values -- otherwise the geomean blows up.  But a value of epsilon is almost as destructive.  I'd change to the arithmetic mean and things should be a lot more stable.  It is much less prone to going to tiny averages due to some tiny values, and as overly small base costs are the problem I think we want to stop using an average that weights small values very heavily (infinitely for 0).

Reply:
Ok, I have changed the code to use the arithmetic mean instead